lib/src/main/java/io/ably/lib/util/JsonUtils.java

```diff
@@ -30,7 +31,10 @@ public JsonUtilsObject add(String key, Object value) {
                 json.addProperty(key, (Number) value);
             } else if (value instanceof JsonUtilsObject) {
                 json.add(key, ((JsonUtilsObject) value).toJson());
+            } else {
+                throw new JsonParseException("Unsupported Object Type");
```

Comment:
I have quickly gone through the reference calls to this method and apart from test classes this seems to be used in `DeviceDetails`s `pushRecipientJsonObject` method. But I don't know whether this might be triggered by a client mistake. If so I think it would be better to. have checked exception, but if this is only something that library code will cause, then unchecked exception is fine

Reply:
This method is used not only in our internal code but also it is used by our clients. The `JsonUtils` is mentioned both in our [README](https://github.com/ably/ably-java#using-the-push-api) and [official docs](https://ably.com/documentation/general/push/publish#channel-broadcast-example). 

Given that, how do we proceed with this issue? Do we want to add a checked exception here which will basically be a change in the API (all clients that use Push API will have to make changes to their code and we probably will need to update the code examples in the docs, right)? Or do we want to get rid of the exception entirely and simply add a log statement informing that a value was not used? :thinking: WDYT @ikbalkaya @QuintinWillison

Reply:
If it's a change in API then that's a 'breaking change', in which case we'll not be able to do that until this SDK makes a release where we can break stuff.

What we can do in the meantime, however, is introduce a new API that does the same thing and deprecate this API. This could be as simple as a new method on `JsonUtils`, or it could be a brand new structure (i.e. class or class+interface set). This would be my preferred approach to solving this.

Reply:
@KacperKluka @ikbalkaya Where are we on this?

Reply:
I'd either change this PR's target branch to the `2.0.0` milestone and make that a breaking change (remembering about updating docs as well) or implement the solution from [this comment](https://github.com/ably/ably-java/pull/729#discussion_r835106748).

Reply:
I am going with "breaking change" as we are moving to v2.0.0 and it seems as the only correct way to do this.

Reply:
I have updated readme https://github.com/ably/ably-java/pull/729/commits/838487b1dfe2c7c2a0c5b855a050cf4983ada18b